### ISSUE TITLE
How do I get the actual treatment variables in the output instead of T0 and T1?

### ISSUE THREAD

**Issue by titubs** [microsoft/EconML#670]
Hi, 
I have this output:

<img width="481" alt="Screen Shot 2022-09-13 at 9 59 55 PM" src="https://user-images.githubusercontent.com/107452474/190063442-196105d1-b608-49d2-bed9-207c530ae104.png">
However, I am not sure what the T0 and T1 map to? My two treatment features are both integers from 0-10. How do I know what the combination of treatment 1 and treatment 2 is which is behind T0 and T1?

**Comment by kbattocchi** [comment]
Could you provide a little bit more context on what method you called and what the arguments were to produce that output?

**Comment by titubs** [comment]
@kbattocchi I guess, I want to first understand how to read this output. Maybe that will answer my question. Does this say that for 1 increase in Treatment 0, my outcome variable will drop by -2.78 on average? If that is so, is that assuming a linear relationship between treatment and outcome? How does that apply to use cases where increasing the price of a product (treatment) has on sales (outcome)? A linear relationship may not be true as there are some customers who may not be as price-sensitive initially but at some point they will...thus, a linear relationship seems odd.

Can you clarify please? That would really help.

**Comment by kbattocchi** [comment]
Most of our estimators assume a linear relationship between the treatment(s) and outcome(s), conditional on the features.  As you note, this may not be realistic, particularly over a wide range of treatments; in that case you may be able to use a transformed version of your treatment and outcome spaces (e.g. using log(quantity) and log(price) would let you estimate a constant price elasticity rather than a linear relationship between quantity and price) or you could a polynomial expansion of your treatment(s) to include higher-order or interaction terms.

However, again to precisely describe how to interpret those results, it would be helpful if you explained how you generated them - is that the output of const_marginal_effect_inference?  If so, it is saying that for whatever the features were in row 0, increasing your first treatment by 1 unit will lower output by ~2.7 units, while increasing the second treatment by 1 unit will lower output by ~14.5 units.

**Comment by titubs** [comment]
@kbattocchi  This makes a lot of sense now. And yes, it is the object causal_forest.const_marginal_effect_inference(X_test) that produced the output of discussion. Thanks a ton!

**Comment by titubs** [comment]
@kbattocchi Lastly, I wanted to check if you could assist with another question related to model performance using EconML Confidence Intervals (for CausalForest for example):

Given the nature of not having ground truth (my understanding is we are computing counter-factuals), how can we measure performance in first place? Am I misunderstanding something here?

**Comment by titubs** [comment]
@kbattocchi Could you help me close the loop on this ticket please? I wanted to confirm my understanding and hope you can help. thanks a ton!

**Comment by kbattocchi** [comment]
Yes, that estimator assumes a constant marginal treatment effect, so the total effect will just be the marginal effect times the change in treatment (25 * (-2.7) in your case).

You are also correct that one fundamental challenge in causal inference is that unlike with predictive models, we do not have ground truth and so we cannot do something like measure performance on a held out data set.  The confidence intervals are computed assuming that the causal model is correctly specified and that your first stage models are good (your first stage models *are* predictive models, and you can see their scores on out-of-sample data using the `nuisance_scores_t` and `nuisance_scores_y` attributes, but that still doesn't tell you if your models are "good enough", because you actually want some unpredictable variation in T that serves as a "found experiment" that lets us learn the treatment effect).

**Comment by titubs** [comment]
@kbattocchi Interesting and makes so much sense! As far as the marginal effect times the change in treatment (25 * (-2.7) in my case goes, is there a way to specify that there is no way that a linear increase is even possible due to point of diminishing returns? I have seen other orthoMLmodels that do something like this (i.e. add a constraint): 

```
monotone_constraints = [-1 if col == T else 0 for col in X+[T]]
model_final = LGBMRegressor(max_depth=3, monotone_constraints=monotone_constraints)
```

Do you have something like this for EconML as well?
